tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Controllers/UserControllerTests.cs

```diff
@@ -0,0 +1,95 @@
+using NHSD.BuyingCatalogue.Organisations.Api.Models;
+using NHSD.BuyingCatalogue.Organisations.Api.ViewModels.OrganisationUsers;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.UnitTests.Controllers
+{
+    [TestFixture]
+    public sealed class UserControllerTests
+    {
+        [Test]
+        public async Task GetUsersByOrganisationId_NoOrganisation_Returns404()
+        {
+            using var controller = new UsersControllerBuilder()
+                .SetOrganisation(null)
+                .Build();
+
+            var result = await controller.GetUsersByOrganisationId(Guid.Empty);
+
+            result.Should().BeOfType<NotFoundResult>();
+        }
+
+        [Test]
+        public async Task GetUsersByOrganisationId_NoUsers_ReturnsEmptyList()
+        {
+            using var controller = new UsersControllerBuilder().Build();
+
+            var result = await controller.GetUsersByOrganisationId(Guid.Empty) as OkObjectResult;
+
+            result.Should().NotBeNull();
+            var users = result.Value as GetAllOrganisationUsersViewModel;
+            users.Should().NotBeNull();
+            users.Users.Should().BeEmpty();
+        }
+
+        [Test]
+        public async Task GetUsersByOrganisationId_ValidId_ReturnsUsers()
+        {
+            var repoUsers = new List<ApplicationUser> { 
+                CreateUser(false),
+                CreateUser(true),
+                CreateUser(false)
+                }.OrderBy(x => x.Id).ToList();
+
+            using var controller = new UsersControllerBuilder()
+                .SetUsers(repoUsers)
+                .Build();
+
+            var result = await controller.GetUsersByOrganisationId(Guid.Empty) as OkObjectResult;
+            
+            result.Should().NotBeNull();
+            var users = result.Value as GetAllOrganisationUsersViewModel;
+            users.Should().NotBeNull();
+
+            var usersSorted = users.Users.OrderBy(x => x.UserId).ToList();
+            AssertEqualUsers(usersSorted, repoUsers);
+        }
+
+        internal static ApplicationUser CreateUser(bool disabled)
+        {
+            return new ApplicationUser
+            {
+                Id = Guid.NewGuid().ToString(),
+                Email = Guid.NewGuid().ToString(),
+                PhoneNumber = Guid.NewGuid().ToString(),
+                FirstName = Guid.NewGuid().ToString(),
+                LastName = Guid.NewGuid().ToString(),
+                Disabled = disabled
+            };
+        }
+
+        internal static void AssertEqualUsers(IEnumerable<OrganisationUserViewModel> actuals, IEnumerable<ApplicationUser> expecteds)
+        {
+            actuals.Count().Should().Be(expecteds.Count());
+
+            for (var i = 0; i < expecteds.Count(); i++)
+            {
+                var actual = actuals.ElementAt(i);
+                var expected = expecteds.ElementAt(i);
+
+                actual.PhoneNumber.Should().Be(expected.PhoneNumber);
```

Comment:
Rather than test each property would it be better to just have one the assert line e.g. `actual.Should().Be(expected)`? Make the code easier to read & also if new properties are added to the object the test would fail.

Reply:
actual.Should().Be.Expected() doesn't play ball, because the database objects differ significantly from the viewmodels. I did try setting up so that would work, but it ended up with a huge amount of code for excluding properties that ApplicationUser has, and converting between parameter names. I don't particularly like this solution, but couldn't think of anything better. I'll ponder it some more

Reply:
One suggestion would be to change the comparison. If the result is a list of viewmodels, create an expected list of viewmodels and compare those rather than two different types.

Reply:
Nice suggestion, I've made this switch